src/native/minipal/thread.h

```diff
@@ -0,0 +1,110 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifndef HAVE_MINIPAL_THREAD_H
+#define HAVE_MINIPAL_THREAD_H
+
+#include <pthread.h>
+#include <stdint.h>
+#include <string.h>
+
+#if defined(__linux__)
+#include <unistd.h>
+#include <sys/syscall.h>
+#elif defined(__FreeBSD__)
+#include <pthread_np.h>
+#elif defined(__NetBSD__)
+#include <lwp.h>
+#elif defined(__HAIKU__)
+#include <kernel/OS.h>
+#endif
+
+#ifdef PTHREAD_MAX_NAMELEN_NP
+#define MAX_THREAD_NAME_SIZE PTHREAD_MAX_NAMELEN_NP
```

Comment:
`PTHREAD_MAX_NAMELEN_NP` includes the null terminator, but `MAX_THREAD_NAME_SIZE` does not.